### PR TITLE
Fixed erroneus failure of extractAllGroupsHorizontal on large columns

### PR DESCRIPTION
**Pull request by Enmk**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Raised the threshold on max number of matches in result of the function `extractAllGroupsHorizontal`.

See tests for a simple example

### CODE REVIEW

File: src/Functions/extractAllGroups.h [hunk 1]
```diff
@@ -173,10 +173,11 @@ class FunctionExtractAllGroups : public IFunction
                         all_matches.push_back(matched_groups[group]);
 
                     /// Additional limit to fail fast on supposedly incorrect usage.
-                    static constexpr size_t MAX_GROUPS_PER_ROW = 1000000;
-
-                    if (all_matches.size() > MAX_GROUPS_PER_ROW)
-                        throw Exception(ErrorCodes::TOO_LARGE_ARRAY_SIZE, "Too large array size in the result of function {}", getName());
+                    static constexpr size_t MAX_MATCHES_PER_ROW = MAX_GROUPS_COUNT * MAX_GROUPS_COUNT;
```
Comment (alexey-milovidov) [inline src/Functions/extractAllGroups.h:176]: Non-obvious - why squared?

Comment (Enmk) [reply src/Functions/extractAllGroups.h:176]: No particular reason, could be anything else. 1M matches per row seemed too high, 128 seemed too low.

Comment (Enmk) [reply src/Functions/extractAllGroups.h]: updated it to be 1000

Comment (alexey-milovidov) [reply src/Functions/extractAllGroups.h]: Ok, let's fix typo and it's ready.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
https://github.com/ClickHouse/ClickHouse/issues/23056

**Pull request merged**